client/ayon_applications/plugins/launcher_actions/debug_shell.py

```diff
@@ -0,0 +1,175 @@
+        if not shell_app:
+            return
+
+        # Get applications
+        applications = self.get_project_applications(
+            application_manager, selection.project_entity)
+        app = self.choose_app(applications, pos)
+        if not app:
+            return
+
+        print(f"Retrieving environment for: {app.full_label}..")
+        env = get_app_environments_for_context(selection.project_name,
+                                               selection.folder_path,
+                                               selection.task_name,
+                                               app.full_name)
+
+        # If an executable is found. Then add the parent folder to PATH
+        # just so we can run the application easily from the command line.
+        exe = app.find_executable()
+        if exe:
+            exe_path = exe._realpath()
+            folder = os.path.dirname(exe_path)
+            print(f"Appending to PATH: {folder}")
+            env["PATH"] += os.pathsep + folder
+
+        cwd = env.get("AYON_WORKDIR")
+        if cwd:
+            print(f"Setting Work Directory: {cwd}")
+
+        print(f"Launching shell in environment of {app.full_label}..")
+        self.launch_app_as_shell(
+            application_manager,
+            shell_app,
+            project_name=selection.project_name,
+            folder_path=selection.folder_path,
+            task_name=selection.task_name,
+            env=env,
+            cwd=cwd)
+
+    @staticmethod
+    def choose_app(
+        applications: list[Application],
+        pos: QtCore.QPoint,
+        show_variant_name_only: bool = False
+    ) -> Optional[Application]:
+        """Show menu to choose from list of applications"""
+        menu = QtWidgets.QMenu()
+        menu.setAttribute(QtCore.Qt.WA_DeleteOnClose)  # force garbage collect
+        menu.setStyleSheet(load_stylesheet())
+
+        # Sort applications
+        applications.sort(key=lambda item: item.full_label)
+
+        for app in applications:
+            label = app.label if show_variant_name_only else app.full_label
+            menu_action = QtWidgets.QAction(label, parent=menu)
+            icon = get_application_qt_icon(app)
+            if icon:
+                menu_action.setIcon(icon)
+            menu_action.setData(app)
+            menu.addAction(menu_action)
+
+        result = menu.exec_(pos)
+        if result:
+            return result.data()
+
+    @staticmethod
+    def get_project_applications(application_manager: ApplicationManager,
+                                 project_entity: dict) -> list[Application]:
+        """Return the enabled applications for the project"""
+        # Filter to apps valid for this current project, with logic from:
+        # `ayon_core.tools.launcher.models.actions.ApplicationAction.is_compatible`  # noqa
+        applications = []
+        for app_name in project_entity["attrib"].get("applications", []):
```

Comment:
Function `get_applications_for_context` should be used.

Reply:
Thanks, done with https://github.com/ynput/ayon-applications/pull/12/commits/6d139e86ca2bce0a7939e6a4f7f0521466ca65d1